src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/PreferStreamAsyncMemoryOverloads.cs

```diff
@@ -0,0 +1,309 @@
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics =>
+            ImmutableArray.Create(PreferStreamReadAsyncMemoryOverloadsRule, PreferStreamWriteAsyncMemoryOverloadsRule);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+            context.RegisterCompilationStartAction(AnalyzeCompilationStart);
+        }
+
+        private void AnalyzeCompilationStart(CompilationStartAnalysisContext context)
+        {
+            // Find the essential type for this analysis
+            if (!context.Compilation.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemIOStream, out INamedTypeSymbol? streamType))
+            {
+                return;
+            }
+
+            // Find the types for the rule message, available since .NET Standard 2.1
+            if (!context.Compilation.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemReadOnlyMemory1, out INamedTypeSymbol? readOnlyMemoryType))
+            {
+                return;
+            }
+            if (!context.Compilation.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemMemory1, out INamedTypeSymbol? memoryType))
+            {
+                return;
+            }
+
+            // Find the additional types for this analysis
+            if (!context.Compilation.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemThreadingCancellationToken, out INamedTypeSymbol? cancellationTokenType))
+            {
+                return;
+            }
+
+            INamedTypeSymbol byteType = context.Compilation.GetSpecialType(SpecialType.System_Byte);
+            if (byteType == null)
+            {
+                return;
+            }
+
+            INamedTypeSymbol int32Type = context.Compilation.GetSpecialType(SpecialType.System_Int32);
+            if (int32Type == null)
+            {
+                return;
+            }
+
+            // Create the arrays with the exact parameter order of the undesired methods
+            var undesiredParameters = new[]
+            {
+                ParameterInfo.GetParameterInfo(byteType, isArray: true, arrayRank: 1), // byte[] buffer
+                ParameterInfo.GetParameterInfo(int32Type),                             // int offset
+                ParameterInfo.GetParameterInfo(int32Type),                             // int count
+            };
+
+            var undesiredParametersWithCancellationToken = new[]
+            {
+                ParameterInfo.GetParameterInfo(byteType, isArray: true, arrayRank: 1),
+                ParameterInfo.GetParameterInfo(int32Type),
+                ParameterInfo.GetParameterInfo(int32Type),
+                ParameterInfo.GetParameterInfo(cancellationTokenType)
+            };
+
+            // Create the arrays with the exact parameter order of the undesired methods
```

Comment:
```suggestion
            // Create the arrays with the exact parameter order of the desired methods
```

Reply:
Am I miss something, actually I just noticed this and below variables are not used, please remove

Reply:
removed, thanks